Sources/ArcGISToolkit/Components/Offline/OfflineManager.swift

```diff
@@ -136,16 +211,22 @@ public class OfflineManager: ObservableObject {
             // Wait for job to finish.
             let result = await job.result
             
-            // Remove completed job from JobManager.
-            Logger.offlineManager.debug("Removing completed job from job manager")
-            jobManager.jobs.removeAll { $0 === job }
-            
-            // This isn't strictly required, but it helps to get the state saved as soon
-            // as possible after removing a job instead of waiting for the app to be backgrounded.
-            jobManager.saveState()
+            if isUsingJobManager {
+                // Remove completed job from JobManager.
+                Logger.offlineManager.debug("Removing completed job from job manager")
+                jobManager.jobs.removeAll { $0 === job }
+                
+                // This isn't strictly required, but it helps to get the state saved as soon
+                // as possible after removing a job instead of waiting for the app to be backgrounded.
+                jobManager.saveState()
```

Comment:
Perhaps `JobManager.jobs` should be private and it should have methods for modifying its jobs? For example, it could have a `removeJobs` method that saves the state after the jobs are removed.